### PR TITLE
Groestlcoin (GRS) support and tests

### PR DESCRIPTION
**Pull request by yura-pakhuchiy**
Adds BIP44 constant, TX api and tests. Bitcoin tests do not fully cover GRS because GRS uses some hashes differently from Bitcoin and changes to other coins may accidentally break GRS.

### CODE REVIEW

File: tests/test_msg_signtx_grs.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# This file is part of the TREZOR project.
```
Comment (keepkeyjon) [inline tests/test_msg_signtx_grs.py]: I don't see these files in https://github.com/trezor/python-trezor ... are they brand new?

Comment (yura-pakhuchiy) [reply tests/test_msg_signtx_grs.py]: Yes, they are based on tests/test_msg_signtx.py from this repo and on test I made for T2 https://github.com/trezor/trezor-core/blob/master/tests/test_apps.wallet.signtx_grs.py

Comment (keepkeyjon) [reply tests/test_msg_signtx_grs.py]: ahh

File: tests/test_msg_signtx_grs.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                proto.ButtonRequest(code=proto_types.ButtonRequest_SignTx),
+                proto.TxRequest(request_type=proto_types.TXINPUT, details=proto_types.TxRequestDetailsType(request_index=0)),
+                proto.TxRequest(request_type=proto_types.TXOUTPUT, details=proto_types.TxRequestDetailsType(request_index=0)),
+                proto.TxRequest(request_type=proto_types.TXOUTPUT, details=proto_types.TxRequestDetailsType(request_index=0)),
+                proto.TxRequest(request_type=proto_types.TXFINISHED),
+            ])
+
+            (signatures, serialized_tx) = self.client.sign_tx('Groestlcoin', [inp1, ], [out1, ])
+
+        self.assertEqual(binascii.hexlify(serialized_tx), '01000000014a9d1fdba915e0907ab02f04f88898863112a2b4fdcf872c7414588c47c874cb000000006a47304402201fb96d20d0778f54520ab59afe70d5fb20e500ecc9f02281cf57934e8029e8e10220383d5a3e80f2e1eb92765b6da0f23d454aecbd8236f083d483e9a7430236876101210331693756f749180aeed0a65a0fab0625a2250bd9abca502282a4cf0723152e67ffffffff01a0330300000000001976a914fe40329c95c5598ac60752a5310b320cb52d18e688ac00000000')
```
Comment (keepkeyjon) [inline tests/test_msg_signtx_grs.py]: Why is the serialized tx different?
https://github.com/trezor/trezor-core/blob/edb8177f1a6a5d83099a540767dda3d29120c83b/tests/test_apps.wallet.signtx_grs.py#L79

Comment (keepkeyjon) [reply tests/test_msg_signtx_grs.py]: ohh, nevermind. that's the prevtx

**Pull request merged**